pkg/controllers/work/applier_fail_if_exists.go

```diff
@@ -0,0 +1,140 @@
+/*
```

Comment:
@ryanzhang-oss Chen raised a valid point. The current API is more like enum. technically, it could have 2*2 combinations. We only support 2 types of them. 

For example, if we support allow co-own & client-side apply, the troubleshooting will be nightmare and the result is hard to predict.

In addition, i cannot think of a case why user does not want co-own if using server-side apply, as they can resolve the conflicts by using field manager.

WDYT? Will the current API confuse users? Not sure if we should call out the allow co-own explicitly in the API or not.